### PR TITLE
Using parent map for accurate computation of balance changes in LTS endpoints

### PR DESCRIPTION
**Pull request by jakrawcz-rdx**
Finally, a feature enabled by https://github.com/radixdlt/babylon-node/pull/511.

The uncommented asserts in the test started passing immediately after this impl, which is a bit suspicious.

### CODE REVIEW

File: core-rust/core-api-server/src/core_api/conversions/lts.rs [hunk 1]
```diff
@@ -54,117 +59,59 @@ pub fn to_api_lts_committed_transaction_outcome(
...
-                if biggest_xrd_debit.is_none() || balance_change < biggest_xrd_debit.unwrap().0 {
-                    biggest_xrd_debit = Some((balance_change, *entity_address));
-                }
-                net_xrd_balance_change += balance_change;
-            }
-        }
+    let mut fee_balance_changes = index_map_new();
+    let records = database.batch_get_ancestry(fee_payments.keys());
+    for (paid_fee_amount_xrd, ancestry) in fee_payments.values().zip(records) {
+        let ancestry = ancestry.expect("a vault must be owned by an account");
```
Comment (dhedey) [inline core-rust/core-api-server/src/core_api/conversions/lts.rs:70]: We typically use `MappingError` instead of panics here, so the error gets turned into a 500. (Because a Core API issue/invariant is imo not sufficiently fatal to take down a node).

Comment (dhedey) [reply core-rust/core-api-server/src/core_api/conversions/lts.rs:70]: Also, the expect message is incorrect - "a vault must be owned" - but not necessarily by an account! It could be some other global entity.

File: core-rust/core-api-server/src/core_api/conversions/lts.rs [hunk 2]
```diff
@@ -54,117 +59,59 @@ pub fn to_api_lts_committed_transaction_outcome(
...
+        let account_address = GlobalAddress::new_or_panic(ancestry.root.0 .0);
```
Comment (dhedey) [inline core-rust/core-api-server/src/core_api/conversions/lts.rs:71]: Again, not necessarily an account, should probably say `global_ancestor_address`

File: core-rust/core-api-server/src/core_api/conversions/lts.rs [hunk 3]
```diff
@@ -54,117 +59,59 @@ pub fn to_api_lts_committed_transaction_outcome(
...
-                            if *resource_address == RADIX_TOKEN {
-                                let fungible_balance_change =
-                                    get_fungible_balance(balance_change).unwrap();
-                                let non_fee_balance_change =
-                                    fungible_balance_change - assumed_fee_balance_change;
-                                if non_fee_balance_change == Decimal::ZERO {
-                                    return None;
-                                }
-                                return Some(to_api_lts_fungible_resource_balance_change(
+    Ok(balance_changes
```
Comment (dhedey) [inline core-rust/core-api-server/src/core_api/conversions/lts.rs:78]: I don't believe this is quite right. If a fee_balance_change is offset by an equal and opposite transfer in, then the balance maybe won't be in the `balance_changes`.

I would suggest the simplest is to make `non_fee_balance_changes = balance_changes` and add the `paid_fee_amount_xrd` XRD fees to the `non_fee_balance_changes` alongside line 72.

The logic below then also becomes a lot simpler - basically just filtering out 0s and mapping.

**Review by dhedey — CHANGES_REQUESTED** [review]
Nice! Glad the tests passed so simply. A few suggestions for tweaks and a minor edge case that I think could do with fixing up (but I might have misunderstood something!)

File: core-rust/core-api-server/src/core_api/conversions/lts.rs (hunk 3 above)
Comment (dhedey) [reply core-rust/core-api-server/src/core_api/conversions/lts.rs:78]: So I had to refactor all of this as part of pulling through damson/elm, so I suggest probably wanting to close this PR and put these changes on top of the elm PR

Comment (jakrawcz-rdx) [reply core-rust/core-api-server/src/core_api/conversions/lts.rs:78]: ok, let me do it 👍 (I assume "the elm PR" == https://github.com/radixdlt/babylon-node/pull/529)
